### PR TITLE
:heart_eyes: disable mobile mode in POS

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

File: pos_mobile/models/pos_config_model.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2019 Dinar Gabbasov <https://it-projects.info/team/GabbasovDinar>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models, fields
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    mobile_version_is_active = fields.Boolean(string="Mobile version is Active", default=True)
```
Comment (ilmir-k) [inline pos_mobile/models/pos_config_model.py]: @GabbasovDinar 
I think it's better to name it ``Mobile version`` instead

Comment (yelizariev) [reply pos_mobile/models/pos_config_model.py]: ``allow_mobile`` + Update description so it's easier to understand Meaning (It's allow mode, not a force mode)

**Review by yelizariev — CHANGES_REQUESTED** [review]
Use proper variable name and clarify docs

File: pos_mobile/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        if not pos_sessions:
+            return werkzeug.utils.redirect('/web#action=point_of_sale.action_client_pos_menu')
+        pos_sessions.login()
+        if pos_sessions.config_id.mobile_version_is_active:
+            session_info = request.env['ir.http'].session_info()
+            session_info['is_mobile'] = True
+            context = {
+                'session_info': json.dumps(session_info),
+            }
+            return request.render('point_of_sale.index', qcontext=context)
```
Comment (yelizariev) [inline pos_mobile/controllers/main.py]: It's better call super and update context in response
See https://github.com/odoo/odoo/blob/10.0/addons/website_portal_sale/controllers/main.py#L14

File: pos_mobile/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            ('state', '=', 'opened'),
+            ('user_id', '=', request.session.uid),
+            ('rescue', '=', False)])
+
+        if not pos_sessions:
+            return werkzeug.utils.redirect('/web#action=point_of_sale.action_client_pos_menu')
+        pos_sessions.login()
+        if pos_sessions.config_id.mobile_version_is_active:
+            session_info = request.env['ir.http'].session_info()
+            session_info['is_mobile'] = True
```
Comment (yelizariev) [inline pos_mobile/controllers/main.py]: If ``mobile_version_is_active``, then it doesn't mean that system *is mobile*, it's only *allow mobile*

File: pos_mobile/views/pos_mobile_template.xml [hunk 4]
```diff
@@ -50,7 +50,7 @@ var MobileCheck = function MobileCheck() {
 })()
 
 // exclude pos_mobile from module list if the url doesn't have parameter "m" in agrs (e.g. /pos/web?m=1 )
```
Comment (yelizariev) [inline pos_mobile/views/pos_mobile_template.xml]: This comment is obsolete and need to be updated

File: pos_mobile/views/pos_mobile_view.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='iface_big_scrollbars']/../.." position="after">
+                    <div class="col-xs-12 col-md-6 o_setting_box" >
+                        <div class="o_setting_left_pane">
+                            <field name="mobile_version_is_active"/>
+                        </div>
+                        <div class="o_setting_right_pane">
+                            <label for="mobile_version_is_active"/>
+                            <div class="text-muted">
+                                Disable or Enable Mobile Version
```
Comment (yelizariev) [inline pos_mobile/views/pos_mobile_view.xml]: You cannot *Enable* Mobile via this option. You can only *Allow*

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Comment by yelizariev** [comment]
More flexible solution might be this:

     auto_mobile = fields.Boolean ("Auto Mobile View", help="Switch to Mobile  view automatically depending on device", default=True)

File: pos_mobile/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import json
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.point_of_sale.controllers.main import PosController
+
+
+class ControllerPos(PosController):
+
+    @http.route('/pos/web', type='http', auth='user')
```
Comment (yelizariev) [inline pos_mobile/controllers/main.py]: ``@http.route()`` is enough

File: pos_mobile/static/src/js/mobile.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ odoo.define('pos_mobile.mobile', function (require) {
     var PosModelSuper = models.PosModel.prototype;
     models.PosModel = models.PosModel.extend({
         initialize: function (session, attributes) {
-            this.is_mobile = odoo.is_mobile;
+            this.is_mobile = odoo.is_mobile && odoo.session_info.allow_mobile;
```
Comment (yelizariev) [inline pos_mobile/static/src/js/mobile.js]: It evaluate to False when auto_mobile is False, but ``m=1``

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Review by yelizariev — APPROVED** [review]

**Comment by yelizariev** [comment]
Change tag to :zap: (new feature) please

**Pull request merged**